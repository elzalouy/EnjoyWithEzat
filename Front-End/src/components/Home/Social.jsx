import { Fragment } from "react";
import { useSelector } from "react-redux";

const Social = () => {
  // I'm here calling the store from redux to get the data i need. It could be editable someday.
  const social = useSelector((state) => state.Person.social);
  return (
    <Fragment>
      {social.map((item) => (
        <a key={item.name} href={item.link} target="blank">
          <i className={"col-1 p-2 bi " + item.icon}></i>
        </a>
      ))}
    </Fragment>
  );
};

export default Social;