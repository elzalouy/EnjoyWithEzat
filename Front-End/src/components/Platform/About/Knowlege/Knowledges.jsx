import { Fragment } from "react";
import { useSelector } from "react-redux";

const Knowledges = () => {
  const knowledges = useSelector((state) => state.Person.knowledges);
  return (
    <Fragment>
          <div className="row w-100 m-0 p-0">
            <div className="col-12 text-left pb-3">Knowledge</div>
            {knowledges && knowledges.map((item) => (
              <div className="col-12 text-left ph-2 pb-2 gray" key={item} id={item}>
                <i className="fas fa-check small-2 bolder gold pr-2"></i>
                {item}
              </div>
            ))}
          </div>
          <div className="col-10 gray line mv-1"></div>
    </Fragment>
  );
};

export default Knowledges;
