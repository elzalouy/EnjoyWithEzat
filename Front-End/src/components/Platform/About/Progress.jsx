import { Fragment } from "react";
import { useSelector } from "react-redux";
import ProgressBar from "../../UIs/progressBar/ProgressBar";
const Progress = () => {
  const { languages } = useSelector((state) => state.Person);
  return (
    <Fragment>
      <div className="row w-100 p-0 m-0">
        <div className="col-12 text-left pb-3">Languages</div>
        {languages.map((lang) => (
          <div className="col-4 p-0 m-0" key={lang.key}>
            <div className="row text-center">
              <ProgressBar value={lang.value} />
              <div className="w-100" id={lang.key}>{lang.key}</div>
            </div>
          </div>
        ))}
      </div>
      <div className="col-10 gray line mv-1 h-100"></div>
    </Fragment>
  );
};

export default Progress;
