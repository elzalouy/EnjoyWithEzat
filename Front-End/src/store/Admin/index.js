import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";
import reducers from './reducers';
const AdminSlice = createSlice({
  name: "admin",
  initialState,
  reducers: reducers
});
export const AdminActions=AdminSlice.actions;
export default AdminSlice.reducer;
