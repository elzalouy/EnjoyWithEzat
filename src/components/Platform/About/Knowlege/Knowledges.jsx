import { Fragment } from "react";
import { useSelector } from "react-redux";

const Knowledges = () => {
  const knowledges = useSelector((state) => state.Person.knowledges);
  return (
    <Fragment>
          <div className="row text-left w-100">
            <div className="col-12 text-left ph-2 pb-3">Knowledge</div>
            {knowledges && knowledges.map((item) => (
              <div className="col-12 text-left ph-2 pb-2 gray" key={item}>
                <i className="fas fa-check small-2 bolder gold pr-2"></i>
                {item}
              </div>
            ))}
          </div>
          <div className="col-10 gray line mv-1"></div>
    </Fragment>
  );
};

export default Knowledges;
