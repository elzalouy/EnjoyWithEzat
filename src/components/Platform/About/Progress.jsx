import { useSelector } from "react-redux";
import ProgressBar from "../../UIs/progressBar/ProgressBar";
const Progress = () => {
  const {languages} = useSelector(state=>state.Person);
  return (
    <div className="row w-100 justify-content-center align-items-center">
      <div className="col-12 text-left ph-2 pb-3">Languages</div>
      {
      languages.map((lang) => (
        <div className="col-3 p-0 m-0 text-center justify-content-center align-items-center w-100" key={lang.key}>
          <ProgressBar value={lang.value}/>
          <div className="text-left w-100">{lang.key}</div>
        </div>
      ))
      }
    </div>
  );
};

export default Progress;
