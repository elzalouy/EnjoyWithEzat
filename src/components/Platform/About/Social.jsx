import { useSelector } from "react-redux";
import classes from "./index.module.css";
const SocialItems = () => {
  const socialItems = useSelector((state) => state.Person.social);
  return (
      <div className={classes.footer + "  bg-3"}>
        {socialItems.map((item) => (
          <a
            key={item.name}
            href={item.link}
            className={"Btn m-0 p-0"}
            target="blank"
          >
            <i className={  "Btn  gray p-2 m-0 bi " + item.icon}></i>
          </a>
        ))}
      </div>
  );
};

export default SocialItems;
