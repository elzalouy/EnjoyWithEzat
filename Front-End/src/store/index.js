import { configureStore } from "@reduxjs/toolkit";
import UiSlice from "./Ui/index";
import PersonSlice from "./Profile/index";
import Projects from './Projects/index'
import Admin from "./Admin/index";
const store = configureStore({
  reducer: { UI: UiSlice, Person: PersonSlice, Projects: Projects, Admin : Admin },
});
export default store;