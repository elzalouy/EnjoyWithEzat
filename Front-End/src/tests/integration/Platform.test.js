import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createMockStore from 'redux-mock-store';
import reduxStore from '../../store';
import { UiActions } from '../../store/Ui';
import MainHeader from '../../components/Platform/Main/Header/Header';
import MainHeaderBrief from '../../components/Platform/Main/Header/Brief';
import AboutInfo from '../../components/Platform/About/Info';
import AboutIntro from '../../components/Platform/About/Intro';
import Knowledges from '../../components/Platform/About/Knowlege/Knowledges';
import Progress from '../../components/Platform/About/Progress';
import Social from '../../components/Platform/About/Social';

const mockstore = createMockStore([]);

describe('Platform > Main > Header', () => {

    const store = mockstore({ UI: reduxStore.getState().UI });
    store.dispatch = jest.fn();
    const component = renderer.create(<Provider store={store}><MainHeader /></Provider>);
    test('should be rendered with redux state', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('should handles opening the about menu', () => {
        renderer.act(
            () => component.root.findByType('li').props.onClick()
        );
        expect(store.dispatch).toBeCalledTimes(1);
        expect(store.dispatch).toBeCalledWith(UiActions.Menu(true));
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('should handles changing the comp to Achievments onClick ', () => {
        renderer.act(() => component.root.findByType('button').props.onClick());
        expect(store.dispatch).toBeCalledTimes(1);
        expect(store.dispatch).toBeCalledWith(UiActions.Switch('achievments'));
    })
});

describe('Platofrm > Main > Brief', () => {
    const store = mockstore({ Person: { yearsExperience: 2, completedProjects: 12 } });
    store.dispatch = jest.fn();
    const component = renderer.create(<Provider store={store}><MainHeaderBrief /></Provider>);
    test("should be rendered with redux state", () => {
        expect(component.toJSON()).toMatchSnapshot();
        const yearsExperienceSpan = component.root.findByProps({ id: 'years' }).children;
        const projectsSpan = component.root.findByProps({ id: 'projects' }).children;
        expect(yearsExperienceSpan).toContain(store.getState().Person.yearsExperience.toString());
        expect(projectsSpan).toContain(store.getState().Person.completedProjects.toString());
    });
})

describe('Platofrm > About > Info', () => {
    const store = mockstore({ Person: { country: 'egypt', city: 'Mansoura', birthday: new Date("nov 7 1996").getFullYear() } });
    const component = renderer.create(<Provider store={store}><AboutInfo /></Provider>);
    test("should be rendered with redux state", () => {
        expect(component.toJSON()).toMatchSnapshot();
        const country = component.root.findByProps({ id: 'country' }).children;
        const age = component.root.findByProps({ id: 'age' }).children;
        const city = component.root.findByProps({ id: 'city' }).children;
        expect(country).toContain(store.getState().Person.country);
        expect(age).toContain(((new Date(Date.now()).getFullYear()) - store.getState().Person.birthday).toString());
    });
})

describe('Platofrm > About > Intro', () => {
    const store = mockstore({ Person: { name: 'ezat elzalouy', jobTitle: 'MERN stack developer' } });
    const component = renderer.create(<Provider store={store}><AboutIntro /></Provider>);
    test("should be rendered with redux state", () => {
        expect(component.toJSON()).toMatchSnapshot();
        const name = component.root.findByProps({ id: 'name' }).children;
        const jobTitle = component.root.findByProps({ id: 'jobTitle' }).children;
        expect(name).toContain(store.getState().Person.name);
        expect(jobTitle).toContain(store.getState().Person.jobTitle);
    });
})
describe('Platofrm > About > Knowledges', () => {
    const store = mockstore({
        Person: {
            knowledges: [
                "Node.js, Express.js, Next.js",
                "NoSqlDB, SqlDB, firebase",
                "React.js, Hooks, Redux",
                "Sass, Less, Bootstrap",
                "UML, SCRUM, Agile",
            ]
        }
    });
    const component = renderer.create(<Provider store={store}><Knowledges /></Provider>)
    test("should be renderd with redux state", () => {
        expect(component.toJSON()).toMatchSnapshot();
        const knowledge = component.root.findByProps({ id: store.getState().Person.knowledges[0] }).children[1];
        expect(knowledge).toEqual(store.getState().Person.knowledges[0]);
    })
})

describe('Platofrm > About > Progress', () => {
    const store = mockstore({
        Person: {

            languages: [
                { key: "Arabic", value: 99 },
                { key: "English", value: 85 },
                { key: "Francais", value: 20 },
            ]
        }
    });
    const component = renderer.create(<Provider store={store}><Progress /></Provider>);
    test("Should be rendered with redux store", () => {
        expect(component.toJSON()).toMatchSnapshot();
        const language = component.root.findByProps({ id: store.getState().Person.languages[0].key });
        expect(language).toBeTruthy();
    });
})
describe('Platofrm > About > Social', () => { 
    const store=mockstore({Person:{social:reduxStore.getState().Person.social}});
    const component=renderer.create(<Provider store={store}><Social/></Provider>);
    test("should be rendere with redux state",()=>{
        expect(component.toJSON()).toMatchSnapshot();
        expect(component.root.findByProps({id:store.getState().Person.social.name})).toBeTruthy();
    })
})