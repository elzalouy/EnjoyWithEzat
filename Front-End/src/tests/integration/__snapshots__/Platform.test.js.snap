// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform > Main > Header should be rendered with redux state 1`] = `
<div
  className="rightSectionHeader col-12 justify-content-start align-items-start"
>
  <div
    className="backCover col-lg"
  />
  <div
    className="HeaderData col-12 p-0"
  >
    <div
      className="d-lg-none text-left justify-content-start nav p-3 m-0"
    >
      <li
        onClick={[Function]}
      >
        <i
          className="fas fa-bars cursor gray hover-gold"
        />
      </li>
    </div>
    <div
      className="header col-12"
    >
      <div
        className="row justify-content-start text-left align-items-center mx-lg-5"
      >
        <div
          className="col-12 m-0 p-0"
        >
          <h1
            className="bolder-1 large-1 my-4"
          >
            Discover my Amazing Art Space!
          </h1>
          <button
            className="Btn px-4 py-2 border-0 bgGold"
            onClick={[Function]}
          >
            EXPOLRE NOW
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Platform > Main > Header should handles opening the about menu 1`] = `
<div
  className="rightSectionHeader col-12 justify-content-start align-items-start"
>
  <div
    className="backCover col-lg"
  />
  <div
    className="HeaderData col-12 p-0"
  >
    <div
      className="d-lg-none text-left justify-content-start nav p-3 m-0"
    >
      <li
        onClick={[Function]}
      >
        <i
          className="fas fa-bars cursor gray hover-gold"
        />
      </li>
    </div>
    <div
      className="header col-12"
    >
      <div
        className="row justify-content-start text-left align-items-center mx-lg-5"
      >
        <div
          className="col-12 m-0 p-0"
        >
          <h1
            className="bolder-1 large-1 my-4"
          >
            Discover my Amazing Art Space!
          </h1>
          <button
            className="Btn px-4 py-2 border-0 bgGold"
            onClick={[Function]}
          >
            EXPOLRE NOW
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Platofrm > About > Info should be rendered with redux state 1`] = `
<div
  className="row bg-2 white p-0 m-0 pt-4 small"
>
  <div
    className="col-6 pt-2 gray"
  >
    Country:
  </div>
  <div
    className="col-6 pt-2 "
    id="country"
  >
    egypt
  </div>
  <div
    className="col-6 pt-2 gray"
  >
    City:
  </div>
  <div
    className="col-6 pt-2 "
    id="city"
  >
    Mansoura
  </div>
  <div
    className="col-6 pt-2 gray"
  >
    Age:
  </div>
  <div
    className="col-6 pt-2 "
    id="age"
  >
    25
  </div>
  <div
    className="col-10 gray line mv-1"
  />
</div>
`;

exports[`Platofrm > About > Intro should be rendered with redux state 1`] = `
Array [
  <h6
    className="white pt-2"
    id="name"
  >
    ezat elzalouy
  </h6>,
  <p
    className="gray small-1 p-0 m-0"
    id="jobTitle"
  >
    MERN stack developer
  </p>,
]
`;

exports[`Platofrm > About > Knowledges should be renderd with redux state 1`] = `
Array [
  <div
    className="row w-100 m-0 p-0"
  >
    <div
      className="col-12 text-left pb-3"
    >
      Knowledge
    </div>
    <div
      className="col-12 text-left ph-2 pb-2 gray"
      id="Node.js, Express.js, Next.js"
    >
      <i
        className="fas fa-check small-2 bolder gold pr-2"
      />
      Node.js, Express.js, Next.js
    </div>
    <div
      className="col-12 text-left ph-2 pb-2 gray"
      id="NoSqlDB, SqlDB, firebase"
    >
      <i
        className="fas fa-check small-2 bolder gold pr-2"
      />
      NoSqlDB, SqlDB, firebase
    </div>
    <div
      className="col-12 text-left ph-2 pb-2 gray"
      id="React.js, Hooks, Redux"
    >
      <i
        className="fas fa-check small-2 bolder gold pr-2"
      />
      React.js, Hooks, Redux
    </div>
    <div
      className="col-12 text-left ph-2 pb-2 gray"
      id="Sass, Less, Bootstrap"
    >
      <i
        className="fas fa-check small-2 bolder gold pr-2"
      />
      Sass, Less, Bootstrap
    </div>
    <div
      className="col-12 text-left ph-2 pb-2 gray"
      id="UML, SCRUM, Agile"
    >
      <i
        className="fas fa-check small-2 bolder gold pr-2"
      />
      UML, SCRUM, Agile
    </div>
  </div>,
  <div
    className="col-10 gray line mv-1"
  />,
]
`;

exports[`Platofrm > About > Progress Should be rendered with redux store 1`] = `
Array [
  <div
    className="row w-100 p-0 m-0"
  >
    <div
      className="col-12 text-left pb-3"
    >
      Languages
    </div>
    <div
      className="col-4 p-0 m-0"
    >
      <div
        className="row text-center"
      >
        <svg
          className="svg"
          height="60"
          width="50"
        >
          <circle
            className="progressRingCircle"
            cx={30}
            cy={30}
            fill="transparent"
            r={22}
            style={
              Object {
                "strokeDasharray": "138.23007675795088 138.23007675795088",
                "strokeDashoffset": 1.3823007675795225,
              }
            }
          />
          <text
            fontSize="10px"
            fontWeight="lighter"
            stroke="white"
            strokeWidth="0.7"
            textAnchor="middle"
            x={25}
            y={30}
          >
            99
            %
          </text>
        </svg>
        <div
          className="w-100"
          id="Arabic"
        >
          Arabic
        </div>
      </div>
    </div>
    <div
      className="col-4 p-0 m-0"
    >
      <div
        className="row text-center"
      >
        <svg
          className="svg"
          height="60"
          width="50"
        >
          <circle
            className="progressRingCircle"
            cx={30}
            cy={30}
            fill="transparent"
            r={22}
            style={
              Object {
                "strokeDasharray": "138.23007675795088 138.23007675795088",
                "strokeDashoffset": 20.734511513692638,
              }
            }
          />
          <text
            fontSize="10px"
            fontWeight="lighter"
            stroke="white"
            strokeWidth="0.7"
            textAnchor="middle"
            x={25}
            y={30}
          >
            85
            %
          </text>
        </svg>
        <div
          className="w-100"
          id="English"
        >
          English
        </div>
      </div>
    </div>
    <div
      className="col-4 p-0 m-0"
    >
      <div
        className="row text-center"
      >
        <svg
          className="svg"
          height="60"
          width="50"
        >
          <circle
            className="progressRingCircle"
            cx={30}
            cy={30}
            fill="transparent"
            r={22}
            style={
              Object {
                "strokeDasharray": "138.23007675795088 138.23007675795088",
                "strokeDashoffset": 110.5840614063607,
              }
            }
          />
          <text
            fontSize="10px"
            fontWeight="lighter"
            stroke="white"
            strokeWidth="0.7"
            textAnchor="middle"
            x={25}
            y={30}
          >
            20
            %
          </text>
        </svg>
        <div
          className="w-100"
          id="Francais"
        >
          Francais
        </div>
      </div>
    </div>
  </div>,
  <div
    className="col-10 gray line mv-1 h-100"
  />,
]
`;

exports[`Platofrm > About > Social should be rendere with redux state 1`] = `
<div
  className="footer  bg-3"
>
  <a
    className="Btn m-0 p-0"
    href="https://www.facebook.com/Eng.ezat.elzalouy/"
    id="facebook"
    target="blank"
  >
    <i
      className="Btn  gray p-2 m-0 bi bi-facebook"
    />
  </a>
  <a
    className="Btn m-0 p-0"
    href="https://twitter.com/Ezat_Elzalouy/"
    id="twitter"
    target="blank"
  >
    <i
      className="Btn  gray p-2 m-0 bi bi-twitter"
    />
  </a>
  <a
    className="Btn m-0 p-0"
    href="https://www.linkedin.com/in/ezat-elzalouy-844bb0133/"
    id="linkedin"
    target="blank"
  >
    <i
      className="Btn  gray p-2 m-0 bi bi-linkedin"
    />
  </a>
  <a
    className="Btn m-0 p-0"
    href="https://github.com/elzalouy"
    id="github"
    target="blank"
  >
    <i
      className="Btn  gray p-2 m-0 bi bi-github"
    />
  </a>
</div>
`;

exports[`Platofrm > Main > Brief should be rendered with redux state 1`] = `
<div
  className="row mt-lg-0 pt-lg-0 mt-5 pt-2 h-auto"
>
  <div
    className="col-sm-3 col-6 small white p-sm-1 mt-5"
  >
    <span
      className="gold bolder  medium"
      id="years"
    >
      2
      +
    </span>
     Years Experience
  </div>
  <div
    className="col-sm-3 col-6 small white p-sm-1 mt-5"
  >
    <span
      className="gold bolder  medium"
      id="projects"
    >
      12
    </span>
     Completed Projects
  </div>
</div>
`;
